tests/framework/helpers/HtmlTest.php

```diff
@@ -140,12 +140,17 @@ public function testCsrfMetaTagsEnableCsrfValidation()
 
     public function testCsrfMetaTagsEnableCsrfValidationWithoutCookieValidationKey()
     {
-        $request = $this->getMock('yii\\web\\Request');
-        $request->method('enableCsrfValidation')->willReturn(true);
-        Yii::$app->set('request', $request);
-        $pattern = '<meta name="csrf-param" content="_csrf">%A<meta name="csrf-token">';
-        $actual = Html::csrfMetaTags();
-        $this->assertStringMatchesFormat($pattern, $actual);
+        $this->mockApplication([
+            'components' => [
+                'request' => [
+                    'class' => 'yii\web\Request',
+                    'enableCsrfValidation' => true,
+                ]
+            ],
+        ]);
+        $this->expectException('yii\base\InvalidConfigException');
+        $this->expectExceptionMessage('yii\web\Request::cookieValidationKey must be configured with a secret key.');
+        Html::csrfMetaTags();
```

Comment:
Isn't assertion with a pattern missing here?

Reply:
No pattern will be generated - exception will be thrown instead.

Reply:
I see now. The mock generates a class that does not work as it normally should.